### ISSUE TITLE
can't load remote rails console

### ISSUE THREAD

**Issue by jongbeau** [sobrinho/heroku-buildpack-pgpool#4]
@sobrinho - thanks for making this buildpack, it looks like it's exactly what I need to get load balancing working on heroku.

I'm experimenting with it and i'm able to get my web severs and sidekiq workers running from the Procfile, however when I try to load a one-off rails console, it immediately exits out after the console loads. Here's my output:

```
➜  ~ heroku run bin/start-pgpool bundle exec rails c
Running bin/start-pgpool bundle exec rails c on ⬢ api-edge... up, run.2760 (Performance-L)
buildpack=pgpool at=pgpool-enabled
buildpack=pgpool at=config-gen-start
WARNING: pid 15: fail_over_on_backend_error is changed to failover_on_backend_error
WARNING: pid 20: fail_over_on_backend_error is changed to failover_on_backend_error
buildpack=pgpool at=config-gen-override DATABASE_URL
buildpack=pgpool at=pgpool-launched pid=23 signal=SIGINT
buildpack=pgpool at=pgpool-start
buildpack=pgpool at=app-launched pid=24
buildpack=pgpool at=app-start
2022-03-06 22:12:13: pid 25: WARNING:  fail_over_on_backend_error is changed to failover_on_backend_error
2022-03-06 22:12:13: pid 25: DETAIL:  setting failover_on_backend_error has no effect
2022-03-06 22:12:13: pid 25: HINT:  use failover_on_backend_error instead
2022-03-06 22:12:13: pid 25: WARNING:  could not load server certificate file "": No such file or directory
2022-03-06 22:12:13: pid 25: LOG:  Backend status file /tmp/pgpool_status does not exist
2022-03-06 22:12:13: pid 25: LOG:  memory cache initialized
2022-03-06 22:12:13: pid 25: DETAIL:  memcache blocks :64
2022-03-06 22:12:13: pid 25: LOG:  pool_discard_oid_maps: discarded memqcache oid maps
2022-03-06 22:12:13: pid 25: LOG:  Setting up socket for 127.0.0.1:5432
2022-03-06 22:12:13: pid 25: LOG:  find_primary_node_repeatedly: waiting for finding a primary node
2022-03-06 22:12:13: pid 25: LOG:  find_primary_node: primary node is 0
2022-03-06 22:12:13: pid 25: LOG:  find_primary_node: standby node is 1
2022-03-06 22:12:13: pid 129: LOG:  PCP process: 129 started
2022-03-06 22:12:13: pid 25: LOG:  pgpool-II successfully started. version 4.1.4 (karasukiboshi)
2022-03-06 22:12:13: pid 25: LOG:  node status[0]: 1
2022-03-06 22:12:13: pid 25: LOG:  node status[1]: 2
2022-03-06 22:12:22: pid 125: LOG:  pool_reuse_block: blockid: 0
2022-03-06 22:12:22: pid 125: CONTEXT:  while searching system catalog, When relcache is missed
Loading edge environment (Rails 6.0.3.2)

Frame number: 0/16
EDGE [1] pry(main)>
buildpack=pgpool at=app-end
buildpack=pgpool at=exit process=app
buildpack=pgpool at=kill-app pid=24
bin/start-pgpool: line 51: kill: (-24) - No such process
buildpack=pgpool at=wait-app pid=24
buildpack=pgpool at=kill-aux name=pgpool pid=23 signal=SIGINT
buildpack=pgpool at=pgpool-end
```

Am I doing something wrong here?

**Comment by sobrinho** [comment]
I don't see anything obvious in the log output.

That only happens with the rails console? If you try to run for instance `heroku run bin/start-pgpool bundle exec rails r 'sleep(5)'` does it work?

**Comment by jongbeau** [comment]
@sobrinho - The immediate exit happens no matter what command I run, even I I just try to run bash.

I tried this:
`heroku run bin/start-pgpool bundle exec rails r 'puts "hello"'`
and it did output `hello`, but it also exited immediately after.

**Comment by sobrinho** [comment]
Tried an app from scratch and the pgpool failed with a FATAL:

```
buildpack=pgpool at=pgpool-enabled
buildpack=pgpool at=config-gen-start
WARNING: pid 14: fail_over_on_backend_error is changed to failover_on_backend_error
WARNING: pid 14: 0 is outside the valid range for parameter "backend_port0" (1024 .. 65535)
WARNING: pid 19: fail_over_on_backend_error is changed to failover_on_backend_error
WARNING: pid 19: 0 is outside the valid range for parameter "backend_port0" (1024 .. 65535)
WARNING: pid 19: 0 is outside the valid range for parameter "backend_port1" (1024 .. 65535)
buildpack=pgpool at=config-gen-override DATABASE_URL
buildpack=pgpool at=pgpool-launched pid=22 signal=SIGINT
buildpack=pgpool at=pgpool-start
buildpack=pgpool at=app-launched pid=23
buildpack=pgpool at=app-start
2022-03-07 16:00:35: pid 24: WARNING:  fail_over_on_backend_error is changed to failover_on_backend_error
2022-03-07 16:00:35: pid 24: DETAIL:  setting failover_on_backend_error has no effect
2022-03-07 16:00:35: pid 24: HINT:  use failover_on_backend_error instead
2022-03-07 16:00:35: pid 24: FATAL:  0 is outside the valid range for parameter "backend_port0" (1024 .. 65535)
buildpack=pgpool at=pgpool-end
buildpack=pgpool at=exit process=pgpool
buildpack=pgpool at=kill-app pid=23
bin/start-pgpool: line 51: kill: (-23) - No such process
buildpack=pgpool at=wait-app pid=23
Loading production environment (Rails 7.0.2.2)
Switch to inspect mode.

buildpack=pgpool at=app-end
buildpack=pgpool at=kill-aux name=pgpool pid=22 signal=SIGINT
bin/start-pgpool: line 62: kill: (22) - No such process
```

I will check what happened, probably pgpool2 made a backwards incompatible change in the conf file and now it is broken. Working on it.

**Comment by jongbeau** [comment]
Interesting. I was able to get pgpool to run (not seeing that FATAL), but it just exits immediately after running the command passed in. Though it works correctly in the Procfile.

**Comment by sobrinho** [comment]
It was my mistake, used wrong variables here.

So far I can tell you can't use a interactive shell with this buildpack due to the way it was done to start the pgpool, same issue of https://github.com/heroku/heroku-buildpack-pgbouncer/issues/22

Although, I'm trying to find a way since it is definitely possible to do so on Bash.

**Comment by sobrinho** [comment]
@jongbeau would you try the branch https://github.com/sobrinho/heroku-buildpack-pgpool/tree/goco-io?

**Comment by sobrinho** [comment]
Fixed by #5
